src/utils/ContractUtils.ts

```diff
@@ -82,3 +91,48 @@ export async function calculateEstimatedGas(
     .estimateGas({ from }, (err, estGas) => (err ? GASLIMIT_DEFAULT : estGas))
   return estimatedGas
 }
+
+/**
+ * Send the transation on chain
+ * @param {string} from account that calls the function
+ * @param {any} estGas estimated gas for the transaction
+ * @param {Web3} web3 web3 objcet
+ * @param {Function} functionToEstimateGas function that we need to send
+ * @param {...any[]} args arguments of the function
+ * @return {Promise<any>} transaction receipt
+ */
+export async function sendTx(
+  from: string,
+  estGas: any,
+  web3: Web3,
+  config: Config,
+  functionToSend: Function,
+  ...args: any[]
```

Comment:
this is not so helpful but not sure we can figure out more precise typings here? As those typings imply I can send whatever after the first params

Reply:
well, they do it like that too, `any` is the solution :-) 

https://github.com/balancer-labs/frontend-v2/blob/develop/src/lib/utils/balancer/web3.ts#L28-L104

Reply:
hmm interesting will take a look at their approach maybe we can "inspire" a little and update the `catch` block

Reply:
So basically our approach is quite different from the balancer one.
 - Firstly they call the send method using EIP1559 params and as a fallback they go back to trying the legacy one. On the other hand we first call web3js lib to get the fee history and call the send method only once. The `getFeeHistory` throws an error if the networks does not support EIP1559 transactions, and so we set the params for the legacy transaction instead of the EIP1559 transaction
 - Also they make https calls to various api's to get the fees and we get them from the web3js lib directly

Reply:
Good research, thanks! Then it looks like we actually have the better solution 😀

Could be fine assuming if getting fee history fails, network doesn't support EIP 1559. Just thought that we might also get some dedicated response text in that case we can work with, like Balancer does, as they really isolate that error case.

So should be fine for now how we do it, we could do more once anybody actually runs into issues